pkg/storage/unified/sql/backend.go

```diff
@@ -650,16 +649,23 @@ func (b *backend) poll(ctx context.Context, grp string, res string, since int64,
 // in a single roundtrip. This would reduce the latency of the operation, and also increase the
 // throughput of the system. This is a good candidate for a future optimization.
 func resourceVersionAtomicInc(ctx context.Context, x db.ContextExecer, d sqltemplate.Dialect, key *resource.ResourceKey) (newVersion int64, err error) {
-	// TODO: refactor this code to run in a multi-statement transaction in order to minimize the number of round trips.
-	// 1 Lock the row for update
-	rv, err := dbutil.QueryRow(ctx, x, sqlResourceVersionGet, sqlResourceVersionRequest{
+	// 1. Increase the resource version to the current epoch with microseconds precision.
+	// This is an update statement that will lock to row and prevent concurrent updates until the transaction is committed.
+	out, err := dbutil.Exec(ctx, x, sqlResourceVersionInc, sqlResourceVersionRequest{
 		SQLTemplate:     sqltemplate.New(d),
 		Group:           key.Group,
 		Resource:        key.Resource,
 		resourceVersion: new(resourceVersion),
 	})
+	if err != nil {
+		return 0, fmt.Errorf("increase resource version: %w", err)
+	}
+	rows, err := out.RowsAffected()
+	if err != nil {
+		return 0, fmt.Errorf("get rows affected: %w", err)
+	}
```

Comment:
> We are using RowsAffected in quite a few places in grafana

Great! Then we're probably cool, I'll just mark as resolved to unblock since this may probably be paranoid anyway.